projects/plugins/boost/app/lib/class-speed-score.php

```diff
@@ -130,11 +155,73 @@ public function fetch_speed_score( $request ) {
 
 		// If this is a fresh install, there might not be any speed score history. In which case, we want to fetch the initial scores.
 		$history = new Speed_Score_History( $url );
-		if ( null === $history->latest() ) {
+		if ( null === $history->latest() && empty( $score_request ) ) {
 			return $this->dispatch_speed_score_request( $request );
 		}
 
-		return $this->prepare_speed_score_response( $url, $score_request );
+		return $this->prepare_speed_score_response( $url, $score_request, $score_request_no_boost );
+	}
+
+	/**
+	 * If it is time to fetch the score without boost, fetch it.
+	 *
+	 * @param string $url Url of the site.
+	 *
+	 * @return Speed_Score_Request
+	 */
+	private function maybe_dispatch_previous_score_request( $url ) {
+
+		// Allow `jb-disable-module` URL param to fetch score without boost modules being active.
+		add_filter( 'jetpack_boost_excluded_query_parameters', array( $this, 'allow_jb_disable_module' ) );
+
+		$url_no_boost = $this->get_boost_modules_disabled_url( $url );
+
+		$history       = new Speed_Score_History( $url_no_boost );
+		$latest_score  = $history->latest( 0, true );
+		$score_request = $this->get_score_request_by_url( $url_no_boost );
+
+		// Refetch the score without boost if it is older than a day.
+		if ( ( empty( $score_request ) || ! $score_request->is_pending() ) && ( null === $latest_score || $latest_score['timestamp'] < strtotime( '- 24 hours' ) ) ) {
+			$score_request = new Speed_Score_Request( $url_no_boost ); // Dispatch a new speed score request to measure score without boost.
+			$score_request->store( 1800 ); // Keep the request for 30 minutes even if no one access the results.
+
+			// Send the request.
+			$response = $score_request->execute();
+
+			if ( is_wp_error( $response ) ) {
+				return $response;
+			}
+		}
+		remove_filter( 'jetpack_boost_excluded_query_parameters', array( $this, 'allow_jb_disable_module' ) );
+
+		return $score_request;
+	}
+
+	/**
+	 * Get Speed_Score_Request instance by url.
+	 *
+	 * @param string $url Url to get the Speed_Score_Request instance for.
+	 *
+	 * @return Speed_Score_Request
+	 */
+	private function get_score_request_by_url( $url ) {
+		return Speed_Score_Request::get(
+			Speed_Score_Request::generate_cache_id_from_url( $url )
+		);
+	}
+
+	/**
+	 * Add query parameters to the url that would disable all boost modules.
+	 *
+	 * @param string $url The original URL we are measuring for score.
+	 *
+	 * @return string
+	 */
+	private function get_boost_modules_disabled_url( $url ) {
+		$modules     = Jetpack_Boost::MODULES;
```

Comment:
I was thinking something like ENABLED_MODULES would be more desirable here. However, existing constants do not work here. So we are good.

Reply:
> In fact, since this URL is used as a key I'm almost tempted to suggest we add support for jb-disable-modules=all to allow a consistent URL even if we release future versions of boost with more modules.

That sounds like a good idea!